graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+
+public type StudentOrTeacher Student|Teacher;
+
+service /graphql on new graphql:Listener(4000) {
+    resource function get profile(int purity) returns StudentOrTeacher {
+        if (purity < 90) {
+            return new Student(1, "Jesse Pinkman");
+        } else {
+            return new Teacher(737, "Walter White", "Chemistry");
+        }
+    }
+}
+
+distinct service class Student {
+    private int id;
+    private string name;
+    
+    public function init(int id, string name) {
+        self.id = id;
+        self.name = name;
+    }
+    
+    resource function get id() returns int {
+        return self.id;
+    }
+    
+    resource function get name() returns string {
+        return self.name;
+    }
+}
+
+distinct service class Teacher {
+    private int id;
+    private string name;
+    private string subject;
+    
+    public function init(int id, string name, string subject) {
+        self.id = id;
+        self.name = name;
+        self.subject = subject;
+    }
+    
+    resource function get id() returns int {
+        return self.id;
+    }
+    
+    resource function get name() returns string {
+        return self.name;
+    }
+        
+    resource function get subject() returns string {
+        return self.subject;
+    }
+}
+```
+
+This will generate the following schema: 
+
+```
+type Query {
+    profile(purity: Int!): StudentOrTeacher!
+}
+
+type Student {
+    id: Int!
+    name: String!
+}
+
+type Teacher {
+    id: Int!
+    name: String!
+    subject: String!
+}
+
+union StudentOrTeacher Student|Teacher
+```
+
+This can be queried using the following document:
+
+```
+query {
+    profile(purity: 75) {
+        ... on Student {
+            name
+        }
+        ... on Teacher {
+            name
+            subject
+        }
+    }
+}
+```
+
+The result will be:
+
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Jesse Pinkman"
+        }
+    }
+}
+``` 
+
+If the following document is used:
+
+```
+query {
+    profile(purity: 99) {
+        ... on Student {
+            name
+        }
+        ... on Teacher {
+            name
+            subject
+        }
+    }
+}
+```
+
+The result will be:
+
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White",
+            "subject": "Chemistry"
+        }
+    }
+}
+``` 
+
+#### Errors (With union)
+A Ballerina GraphQL resource can return an `error` with the union of the types mentioned above.
+
+Note: A resource cannot return `error` or any subtype of `error` alone or, `error?` which will results in a compilation error.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get profile(int id) returns Person|error {
+    if (id == 1) {
+        return { name: "Walter White", age: 51 };
+    } else {
+        return error(string `Invalid ID provided: ${id}`);
+    }
+}
+```
+
+This can be queried using the following document:
```

Comment:
```suggestion
This can be queried using the following document:
```
Is it correct to say "the following document"? Or, should it be "the following code"? Please check all occurrences.

Reply:
This is the general term used in GraphQL. The text used to query the GraphQL endpoint is called the `document`.